ws-server/src/server.rs

```diff
@@ -326,9 +326,8 @@ async fn background_task(
 		// Terminate connection and send close message.
 		let _ = sender.close().await;
 
-		// Force `conn_tx` to this async block and close it down
-		// when the connection closes to be on safe side.
-		close_notify_server_stop.notify_one();
+		// Notify all listeners and close down associated tasks.
+		close_notify_server_stop.notify_waiters();
```

Comment:
this is the fix

Reply:
Ahh okay makes sense. So listeners were being launched from `pipe_from_try_stream`, and in this case because it was `notify_one` only a single task was stopped. 

Good catch 👍

Reply:
yeah, basically every connection can have n number of subscriptions and we want to cancel them all when the connection is killed.

the subscriptions will/may be canceled as soon some message is sent via them but that might take a while so we want to do it as soon as the connection is closed